exasol_sagemaker_extension/autopilot_prediction.py

```diff
@@ -0,0 +1,34 @@
+import os
+import json
+from typing import Callable
+from exasol_sagemaker_extension.autopilot_utils.model_prediction import \
+    AutopilotPrediction
+
+
+class AutopilotPredictionUDF:
+    def __init__(self, exa, model_connection_name,
+                 prediction_method: Callable = AutopilotPrediction.predict):
+        self.exa = exa
+        self.model_connection_name = model_connection_name
+        self.prediction_method = prediction_method
+
+    def __parse_model_connection(self):
+        model_connection = self.exa.get_connection(self.model_connection_name)
+        endpoint_info_json = json.loads(model_connection.password)
+        name = endpoint_info_json['name']
+        status = endpoint_info_json['status']
+        return name, status == 'deployed'
+
+    def run(self, ctx):
+        os.environ["AWS_DEFAULT_REGION"] = ""
+        os.environ["AWS_ACCESS_KEY_ID"] = ""
+        os.environ["AWS_SECRET_ACCESS_KEY"] = ""
+
```

Comment:
we need to have an  exa connection holding sagameker-execution-role priviliges. I think passing this connection name as an argument of PredictionUDF is the best way. any comments?

Reply:
I guess, we need this as well during endpoint creation. If yes, hard code it into the UDF as we do it with the endpoint connection.

Reply:
I thought of this solution as well, but I thought that if users update their connection information before prediction (for example, if they revoke the AWS_KEY that created the endpoint), the prediction wouldn't work.

On the other side, It would also be nice if PredictUDF doesn't take arguments except  columns.

Reply:
yep, if the user needs to revoke the key, they can change the connection object.